### DISCUSSION TITLE
Description editor sidebar integration

### DISCUSSION THREAD

**Discussion by manuelaugustin** [woocommerce/woocommerce#41456] [WooCommerce New Product Editor]
When opening the `Edit description` modal, it's not currently possible to register a sidebar as far as we could see. In the block editor on posts/pages this is possible, and it would be a valuable integration for us as a plugin. Is it correct that it's not possible to integrate in this way at the moment? If so, could you make it possible to integrate here?

<img width="1040" alt="Screenshot_2023-11-15_at_10_06_19" src="https://github.com/woocommerce/woocommerce/assets/27805437/0dfd302d-5648-4222-8445-3828bc136fd9">

**Comment by mattsherman** [comment]
Hello @manuelaugustin!

This is currently not possible. We will be looking into enhancing the extensibility to make this possible.

Also, there will be UX guidelines published soon that will provide more details on best practices for extending the product editor.

**Comment by mattsherman** [comment]
@manuelaugustin We are currently looking into extensibility points for the modal description editor. What sort of things would you want put in a sidebar in the modal description editor vs in a sidebar for the product editor as a whole, if one existed?

Also, have you considered adding your items directly to the product editor form?

**Reply by manuelaugustin** [reply]
@mattsherman Hi, thanks for following up on this! And apologies for my late reply!

In Yoast SEO we offer real-time feedback on SEO and readability as you type your text. See for example the way this currently looks in the block editor.

It's crucial that we can provide feedback to users as they type their text, without users having to switch between different contexts. We also offer highlighting functionality to highlight specific parts of the text, so that is also something that users should be able to do while editing their text. (On a related note, we also asked for access to annotations in a separate [thread](https://github.com/woocommerce/woocommerce/discussions/42761)).

So for us to be able to further work on the integration with the new Woo product editor, it's necessary to have this extensibility point available.

<img width="818" alt="analysis block editor" src="https://github.com/woocommerce/woocommerce/assets/27805437/36e06646-c520-484a-ac23-e243fcf05287">

**Reply by manuelaugustin** [reply]
Hi @mattsherman, could you let me know if there is any update on this? Thanks!

**Reply by mattsherman** [reply]
Thanks for the detailed explanation of Yoast's extensibility needs. I do not have an update currently, but I will be sure to discuss this specific point in an extensibility-related meeting I have today.

**Reply by manuelaugustin** [reply]
Thanks @mattsherman! Are there any updates from the meeting you've had?

**Reply by mattsherman** [reply]
@manuelaugustin The current plan is to work on this in an upcoming development cycle. The plan is to implement this extensibility in a manner such that extensions such as Yoast can adapt the extensibility they are using to hook into the post editor to do the same sort of thing for the sidebar in the description editor.

**Reply by manuelaugustin** [reply]
Hi @mattsherman, could you let me know if there are any news regarding this functionality?

**Reply by mattsherman** [reply]
Hi @manuelaugustin. I started work on implementing changes to support this last week. I do not yet have a fully functional solution, but the current plan is to use `ComplementaryArea` as is done in the post editor to allow for extensions to add their own sidebars.

There is no PR description/test instructions, and it isn't even fully functional yet, but you can subscribe to the following PR if you want to follow along. As I flesh things out more this week it will get to a point where you can begin to take it for a spin yourself and evaluate how it meets your needs. I'll be sure to let you know when I believe the PR is ready for you to try out.

https://github.com/woocommerce/woocommerce/pull/46597

**Reply by manuelaugustin** [reply]
Thanks, that's great! We'll keep an eye on your PR 🙂

**Comment by mattsherman** [comment]
@manuelaugustin I have a PR up for review that adds sidebar extensibility to the description modal editor.

- https://github.com/woocommerce/woocommerce/pull/46597

Block selection and modification works, and you have access to the product being edited (the PR includes a link to a test plugin that demonstrates this).

Any feedback on description modal editor extensibility support/issues is appreciated... the PR has a list of some known issues that I will be following up on in the next week or so.
